python_modules/airline-demo/airline_demo/types.py

```diff
@@ -24,50 +20,41 @@
 )
 
 
-class SparkDataFrameParquetSerializationStrategy(SerializationStrategy):
-    def serialize_value(self, _context, value, write_file_obj):
-        parquet_dir = tempfile.mkdtemp()
-        shutil.rmtree(parquet_dir)
-        value.write.parquet(parquet_dir)
-
-        archive_file_obj = tempfile.NamedTemporaryFile(delete=False)
-        try:
-            archive_file_obj.close()
-            archive_file_path = archive_file_obj.name
-            zipfile_path = shutil.make_archive(archive_file_path, 'zip', parquet_dir)
-            try:
-                with open(zipfile_path, 'rb') as archive_file_read_obj:
-                    write_file_obj.write(archive_file_read_obj.read())
-            finally:
-                if os.path.isfile(zipfile_path):
-                    os.remove(zipfile_path)
-        finally:
-            if os.path.isfile(archive_file_obj.name):
-                os.remove(archive_file_obj.name)
-
-    def deserialize_value(self, context, read_file_obj):
-        archive_file_obj = tempfile.NamedTemporaryFile(delete=False)
-        try:
-            archive_file_obj.write(read_file_obj.read())
-            archive_file_obj.close()
-            parquet_dir = tempfile.mkdtemp()
-            # We don't use the ZipFile context manager here because of py2
-            zipfile_obj = zipfile.ZipFile(archive_file_obj.name)
-            zipfile_obj.extractall(parquet_dir)
-            zipfile_obj.close()
-        finally:
-            if os.path.isfile(archive_file_obj.name):
-                os.remove(archive_file_obj.name)
-        return context.resources.spark.read.parquet(parquet_dir)
+class SparkDataFrameS3StoragePlugin(TypeStoragePlugin):  # pylint: disable=no-init
+    @classmethod
+    def set_object(cls, object_store, obj, _context, _runtime_type, paths):
+        target_path = object_store.key_for_paths(paths)
+        obj.write.parquet('s3a://' + target_path)
+        return target_path
+
+    @classmethod
+    def get_object(cls, object_store, context, _runtime_type, paths):
+        return context.resources.spark.read.parquet('s3a://' + object_store.key_for_paths(paths))
+
+
+class SparkDataFrameFilesystemStorageOverride(TypeStoragePlugin):  # pylint: disable=no-init
```

Comment:
StoragePlugin instead of StorageOverride?

Reply:
yep